---
title: "Amino Acid k-mer Feature Extraction for Quantitative Antimicrobial Resistance (AMR) Prediction by Machine Learning and Model Interpretation for Biological Insights"
collection: publications
permalink: /publications/2020-10-28-AMR
excerpt: 'This paper is about an incremental learning taxonomic classifier.'
date: 2020-10-28
venue: 'Biology'
paperurl: 'https://doi.org/10.3390/biology9110365'
citation: 'ValizadehAslani, T.; Zhao, Z.; Sokhansanj, B.A.; Rosen, G.L. Amino Acid k-mer Feature Extraction for Quantitative Antimicrobial Resistance (AMR) Prediction by Machine Learning and Model Interpretation for Biological Insights. <i>Biology</i> 9, 365 (2020).'
---
## Background
Infectious bacteria (microbes) are able to evolve to become resistant to antibiotics (develop antimicrobial resistance, or AMR). Resistant microbes are harder to treat, requiring higher doses, or alternative medications, which can be more toxic. Because of inappropriate use of medicine, microbes are being subjected to evolutionary pressure resulting in increased AMR development. As a result, AMR is emerging one of the biggest public health challenges of our timeâ€”posing the risk of a pandemic without effective treatment or vaccine.

## Results
In this study, we have proposed a new feature extraction method, counting amino acid k-mers or oligopeptides, which provides easier model interpretation compared to counting nucleotide k-mers and reaches the same or even better accuracy in comparison with different methods. Additionally, we have trained machine learning algorithms using different feature extraction methods and compared the results in terms of accuracy, model interpretability and computational complexity. We have built a new feature selection pipeline for extraction of important features so that new AMR determinants can be discovered by analyzing these features. This pipeline allows the construction of models that only use a small number of features and can predict resistance accurately. We show that our proposed method produces comparable or even more accurate results when compared to existing methods for the goal of dose prediction, and it can provide additional insight for scientists who study AMR mechanisms.

## Conclusions
We have shown that amino acid k-mers are less prone to the problem of large feature size compared to nucleotide k-mers. Moreover, amino acid k-mers are easier to interpret, because their top features are more likely to be uniquely identified using search algorithms such as BLAST. The main drawback of amino acid k-mers is that they require assembly of the genes. In a situation where assembly of the genes is not a problem, we recommend amino acid k-mers.
